python/idsse_common/idsse/common/grid_proj.py

```diff
@@ -0,0 +1,113 @@
+        self._dy = dy if dy else dx
+
+    @classmethod
+    def from_proj_grid_spec(cls, proj_string: str, grid_string: str) -> Self:
+        """ Create GridProj instance from projection grid specs
+
+        Args:
+            proj_string (str): pyproj projection string
+            grid_string (str): Grid string
+
+        Returns:
+            Self: New instance of GridProj which can then be converted to any format
+        """
+        crs = CRS.from_proj4(proj_string)
+        grid_args = {
+            key[1:]: float(value)
+            for key, value in (
+                pair.split('=') for pair in grid_string.split(' ')
+            )
+        }
+        return GridProj(crs,
+                        grid_args['lat_ll'], grid_args['lon_ll'],
+                        int(grid_args['w']), int(grid_args['h']),
+                        grid_args['dx'], grid_args['dy'])
+
+    def map_proj_to_pixel(self, x, y) -> Tuple[float, float]:
+        """Map projection to a pixel"""
+        return self.map_geo_to_pixel(*self._trans.transform(x, y))  # pylint: disable=not-an-iterable
+
+    def map_pixel_to_proj(self, x: float, y: float) -> Tuple[Any, Any]:
+        """Map pixel to a projection"""
+        return self._trans.transform(*self.map_pixel_to_geo(x, y), direction=TransformDirection.INVERSE)
+
+    def map_proj_to_geo(self, x, y) -> Tuple[float, float]:
+        """Map projection to geographic coordinates"""
+        return self._trans.transform(x, y)
+
+    def map_pixel_to_geo(self, x: float, y: float) -> Tuple[float, float]:
+        """Map pixel to geographic coordinates"""
+        return x * self._dx + self._x_offset, y * self._dy + self._y_offset
+
+    def map_geo_to_proj(self, x: float, y: float) -> Tuple[Any, Any]:
+        """Map geographic coordinates to a projection"""
+        return self._trans.transform(x, y, direction=TransformDirection.INVERSE)
+
+    def map_geo_to_pixel(
+        self, x: float,
+        y: float,
+        rounding: Optional[PixelRounding] = None
+    ) -> Tuple[Union[int, float], Union[int, float]]:
+        """Map geographic coordinates to pixel x and y
+
+        Args:
+            x (float): x geographic coordinate
+            y (float): y geographic coordinate
+            rounding (Optional[PixelRounding]):
+                ROUND to apply round() to pixel values, FLOOR to apply math.floor().
+                By default, pixels are not rounded and will be returned as floats
+
+        Returns:
+            Tuple[Union[int, float], Union[int, float]):
+                x, y values for pixel, rounded to ints if rounding parameter passed, otherwise floats
+        """
+        pixel_x = (x - self._x_offset) / self._dx
+        pixel_y = (y - self._y_offset) / self._dy
+
+        if rounding is PixelRounding.ROUND:
+            return (round(pixel_x), round(pixel_y))
+        if rounding is PixelRounding.FLOOR:
+            return (floor(pixel_x), floor(pixel_y))
+        return (pixel_x, pixel_y)
```

Comment:
@Geary-Layne I added a `rounding` argument to map_geo_to_pixel that is either ROUND, FLOOR or None, is this about what you envisioned?